### ISSUE TITLE
Error messages

### ISSUE THREAD

**Issue by emhoracek** [the-learning-collective/tlcjs#3]
As I've been writing sample code, I've gotten a bunch of misleading or confusing errors, and I've also not gotten errors at all for very broken code (forgetting parentheses, mostly). That made things a little difficult for me. I imagine for a beginner it would be much more frustrating!! 

I think we should go through and add error messages to all our public functions when they're invoked with the wrong number or wrong type of arguments.

Maybe like this?

``` (js)
/* emptyScene :: number -> number -> scene */
function emptyScene(width, height) {

  if (arguments.length !== 2 ||
      (typeof arguments[0] !== "number") ||
      (typeof arguments[1] !== "number")) {
    throw new Error("emptyScene(): Requires two arguments, a width and a height, 
                                both numbers. For example: emptyScene(300, 200)")
  }

  return { elements: [],
           width: width,
           height: height };
}
```

**Comment by dbp** [comment]
@emhoracek I think this is a phenomenal idea, and I actually was going to do something like this today, but didn't want to make merge-hell :) But if you want to do this, I won't touch anything until at least tomorrow evening, so work on `master` and change everything! :)

I think we can probably make it shorter though. Write a helper function called `_type` such that we can write what you wrote as:

```
/* emptyScene :: number -> number -> scene */
var emptyScene = _type("emptyScene", ["number", "number"], function(width, height) {
  return { elements: [],
           width: width,
           height: height };
});
```

With `arguments` and `apply()` (or `call`?) it should be totally possible to write that function.

Also, for sanity, we should probably define the types as variables, so we aren't as likely to typo them:

```
var Number = "number";
var String = "string";
...
```

Making the above be like:

```
var emptyScene = _type("emptyScene", [Number, Number], ...
```

And check the type arguments against `undefined` (in case you write `Nmber` instead of `Number`).

**Comment by dbp** [comment]
Oh, I didn't see the example at the end of the error message. Actually, probably `_type` should just take the whole error message, rather than just the name of the function. It's good to write nice error messages :)

**Comment by dbp** [comment]
Oh, nevermind, scratch the testing against `undefined` for the type arguments. An undefined variable throws a `ReferenceError`, so it wouldn't be silent.

**Comment by emhoracek** [comment]
@dbp I wasn't going to do anything right away, I just wanted to note it while I was thinking of it. 

Ooh, an excuse to learn about call() and apply(). :+1:

**Comment by dbp** [comment]
This _should_ be closed by cbf3b711d0205c4702d68bad8d621fc61daad7a9.

We can re-open if it turns out more work needs to be done!
